### ISSUE TITLE
Keep forks with commit to non-default branches

### ISSUE THREAD

**Issue by katrinleinweber** [useful-forks/useful-forks.github.io#33]
1k thanks for this tool :-)

Relevant work on a fork can be hidden in a non-default branch, if for example the author planned to submit a PR properly, but eventually didn't.

Would it be useful here to also check for such branches ([not just](https://github.com/useful-forks/useful-forks.github.io/blame/v0.5/README.md#L10) the [default branch](https://github.com/useful-forks/useful-forks.github.io/blob/v0.5/project/useful-forks.js#L197)), and keep forks in the results list, if any are found?

I've [implemented such logic](https://github.com/katrinleinweber/poPR/blob/main/popr/reduce_to_potential_pr_heads.py) a while ago in a similar tool. While I can't port it to JS, I hope the example is still helpful to consider this idea.

The increased number of API calls [can be mitigated](https://github.com/katrinleinweber/poPR/blame/284cd32e4a60206697a0f6e36b4f3f0ec67848d5/README.md#L35-L36), if necessary.

**Comment by payne911** [comment]
> 1k thanks for this tool :-)

My pleasure! It was born out of personal necessity and curiosity. :)

> Would it be useful here to also check for such branches

Absolutely !

> The increased number of API calls can be mitigated, if necessary.

That is indeed my main concern. For massively forked repositories, I'd rather see more results with a simplified view, than an incomplete but detailed list. And I assume this tool's relevance increases with the amount of forks, so I'm mostly aiming at keeping it this way.

My current plan is to allow people to save their "search preferences" through the little gear icon next to the search input text field, so that would be how I would add the feature of scanning all branches.

The simple act of querying for the list of branches of a fork is an API call. Then comparing branches would be another one. That's pretty much doubling the amount of calls the tool is currently making, hence why I'd keep it as opt-in option.

**Comment by katrinleinweber** [comment]
Yeah, that approach is for a slightly different goal than useful-forks': Assuming that each finding needs to be reviewed manually anyway, so waiting for the full list isn't needed.
 
> … save their "search preferences" …

Also cool, thanks! :-)
